util-flywayinstaller/src/main/java/com/github/cafapi/util/flywayinstaller/Migrator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.util.flywayinstaller;
+
+import com.github.cafapi.util.flywayinstaller.exceptions.InvalidConnectionStringException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.flywaydb.core.Flyway;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Migrator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Migrator.class);
+    private static final String CONNECTION_URL_REGEX = "^.*\\/\\/.+:\\d+\\/";
+    private static final String CREATE_DATABASE = "CREATE DATABASE \"%s\"";
+    private static final String DOES_DATABASE_EXIST
+        = "SELECT EXISTS ( SELECT datname FROM pg_catalog.pg_database WHERE lower( datname ) = lower( ? ) );";
+
+    private Migrator()
+    {
+    }
+
+    public static void migrate(
+        final String connectionString,
+        final String dbName,
+        final String username,
+        final String password
+    ) throws InvalidConnectionStringException, SQLException
+    {
+        logReceivedArgumentsIfDebug(connectionString, dbName, username, password);
+
+        LOGGER.info("Starting migration ...");
+
+        try {
+            final PGSimpleDataSource dbSource = new PGSimpleDataSource();
+            final String urlAfterConversion = checkAndConvertConnectionUrl(connectionString);
+            dbSource.setUrl(urlAfterConversion);
+            dbSource.setUser(username);
+            dbSource.setPassword(password);
+
+            if (!doesDbExist(dbSource, dbName)) {
+                LOGGER.debug("reset or createDB");
+                resetOrCreateDatabase(dbSource, dbName);
+            }
+            // Update dbSource url to include the database name
+            dbSource.setUrl(urlAfterConversion + dbName);
```

Comment:
Can't you use [`.setDatabaseName()`](https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/ds/common/BaseDataSource.html#setDatabaseName-java.lang.String-)?

Reply:
Done + no longer using url but `server, port, dbName, user and pwd`